src/child/containers/sidebars/search/Search.js

```diff
@@ -36,12 +37,12 @@ class Search extends Component {
         }, SEARCH_TIMEOUT_INTERVAL);
     }
 
-    onIconClick(stock) {
-        this.props.dispatch(toggleFavourite(stock.code));
+    onIconClick(stockCode) {
+        this.props.dispatch(toggleFavourite(stockCode));
     }
 
-    onClick(stock) {
-        this.props.dispatch(selectStock(stock.code, stock.name));
+    onClick(stockCode, stockName) {
+        this.props.dispatch(selectStock(stockCode, stockName));
```

Comment:
Might be worthwhile to have a look at keeping the name somewhere and only pass codes around.